library/std/src/lib.rs

```diff
@@ -186,22 +186,12 @@ macro_rules! panic {
     ($msg:expr $(,)?) => ({
         kani::panic(stringify!($msg));
     });
-    // The first argument is the message and the rest contains tokens to be included in the msg.
+    // All other cases, e.g.:
     // `panic!("Error: {}", code);`
-    //
-    // Note: This macro may match things that wouldn't be accepted by the panic!() macro. we have
-    // decided to over-approximate the matching so we can deal with things like macro inside a macro
-    // E.g.:
-    // ```
-    // panic!(concat!("Message {}", " split in two"), argument);
-    // ```
-    // The std implementation of `panic!()` macro is implemented in the compiler and it seems to
-    // be able to do things that we cannot do here.
-    // https://github.com/rust-lang/rust/blob/dc2d232c7485c60dd856f8b9aee83426492d4661/compiler/rustc_expand/src/base.rs#L1197
-    ($msg:expr, $($arg:tt)+) => {{
+    ($($arg:tt)+) => {{
         if false {
-            let _ = format_args!($msg, $($arg)+);
```

Comment:
Any chance you can add a note on why this is here? I was also wondering if rustc will prune this. If not, can we use a noop assignment instead? Something like:

```
let _ = ($($arg)+)
```

Reply:
This line is supposed to make sure that any incorrect usage of `panic`'s formatting arguments is reported by Kani, e.g.
```rust
panic!("{}"); // positional argument but no argument given
```
However, I just found out that it's not doing that :( I'll have to figure out a fix.

Reply:
The error checking seems to be broken for the `assert` macro as well with https://github.com/model-checking/kani/pull/1858 :(

Reply:
Error checking was restored with the switch to using `core::assert`.